### DISCUSSION TITLE
Can't find where to change language to Spanish

### DISCUSSION THREAD

**Discussion by RafaelLinux** [helloSystem/hello#472] [General]
I read in documentation that localization to Spanish is completed, but I find no way to change language once I booted. I clicked in "Desktop" icon and no application is opened. Is there where I should be able to change language?

Thank you

**Comment by probonopd** [comment]
The Language Chooser should come up during initial boot of the Live system if no other language setting has been detected on the system.

The easiest way to get Spanish is to attach a [Spanish Raspberry Pi Keyboard and Hub](https://www.raspberrypi.com/products/raspberry-pi-keyboard-and-hub/). Another way is to have the `7c436110-ab2a-4bbb-a880-fe41995c9f82-prev-lang:kbd` EFI NVRAM on the machine set to `es:8`, which will result in a Spanish keyboard layout and Spanish as the system language to be used. Future versions may do this automatically.

See https://github.com/helloSystem/hello/wiki/EFI-NVRAM for details.

**Reply by RafaelLinux** [reply]
I'm using QEMU in a x64 PC. I booted serveral times with Hellosystem but never a language menu is showed (like other distros does) in booting process. But I expected to find a way to change localization once system was booted. It seems not to be simple.

**Comment by louies0623** [comment]
@probonopd I don't know how many times I have told you to add language settings, but you have always insisted on your own ideas, and there is no setting method that allows users to switch languages ​​easily, insisting on automatic detection, if it is detected that it is not what the user needs  Language, then there is no language setting, the user has to ask you for help until the language setting is added in the system.

The most taboo thing for normal users is that a simple setting needs to be processed with a troublesome terminal.

![macosxdp2](https://user-images.githubusercontent.com/44593430/215298019-72ad480f-7c2c-4be4-b0fc-fbd13a7f2037.png)

**Reply by RafaelLinux** [reply]
Yes, something like that. And even some times user need to change language from native to English to take screenshots and share them in English forums, so it's important to let user change it at least after boot process ;)

**Reply by probonopd** [reply]
It is not implemented yet.

**Reply by louies0623** [reply]
This absolutely needs to be implemented.

**Reply by probonopd** [reply]
Pull requests welcome if anyone sees a strong need for this.
But the values need to be persisted to EFI NVRAM as I don't want each operating system and version to have this stored separately.

For my personal needs, I have a [Raspberry Pi Keyboard and Hub](https://www.raspberrypi.com/products/raspberry-pi-keyboard-and-hub/) attached to my computer; this results in the keyboard layout, the system language, and the timezone to be set accordingly automatically. So I have no need to change anything manually.

**Reply by louies0623** [reply]
Otherwise, like Windows, there is a separate system language pack, if you want to install other languages, install it, "If you want to switch languages, go to the language settings".
Then, for the development system, if there is an updater on each build, even the language is kept, the system is updated with the language and like the Windows updater, you don't have to reset all the time.

**Reply by probonopd** [reply]
Yes, by storing these things in EFI NVRAM, you can switch between multiple SSDs containing different versions of helloSystem, and/or boot different Live ISOs (e.g., from a big SSD using Ventoy) and the settings will always be "there".

**Reply by louies0623** [reply]
Although this method solves your problem, it becomes a bigger problem for others "how to switch or change or remove EFI NVRAM language".

If it is automatically detected by the keyboard, it feels like if my grandma wears a skirt from a different country, she will speak the local language of that country.

**Comment by shedtepherd** [comment]
My mouse froze during the language selection when booting to the live thumb drive and I clicked it and moved it around, but accidentally selected French as my language. Is it possible to change it to English from within the DE?

**Reply by probonopd** [reply]
Uh. That's a situation I hadn't thought of, retroactively changing the system language after it has been manually set....

```
unset LANG
launch Language
```

should bring up the chooser again.

**Reply by kyleblizzard** [reply]
I had this same issue. I would second the value of a language changer in the settings.

**Reply by Flalaski** [reply]
> Uh. That's a situation I hadn't thought of, retroactively changing the system language after it has been manually set....
> 
> ```
> unset LANG
> launch Language
> ```
> 
> should bring up the chooser again.

I have the same issue. it's accidentally in french & I need it in english. 
I set the keyboard to english & that works fine, but it's all still in french. 
I tried those in QTerminal & it says language could not be found ~ could not be launched
Edit: I since restarted the computer & now it's in English, likely thanks to the steps taken. I'm all good now.
I'll third on to adding language changer in settings. it's an accessibility thing
